### DISCUSSION TITLE
Android 13?

### DISCUSSION THREAD

**Discussion by JulGer** [j-kaltes/Juggluco#5] [General]
Hi since the developers of the original libre 3 App supposedly fixed the Android 13 problem, is it possible to do so with Juggluco also? Didn't test the new app version. I am really happy with Juggluco, now that we don't *need* the original Abbott App anymore and I'm still on Android 12, but would be really great to know that I could upgrade my Android anytime if needed...😉

**Comment by j-kaltes** [comment]
When you look at what has changed between 3.4.0 and 3.4.2, you find the following in AndroidManifast.xml:
```
 <receiver android:exported="false" android:name="com.adc.trident.app.frameworks.mobileservices.libre3.libre3BLESensor">
            <intent-filter>
                <action android:name="com.freestylelibre3.app.de.action.BLE_CONNECT_RETRY"/>
            </intent-filter>
 </receiver>
```

So it seems that they have a timer that connects Bluetooth after a certain period of time; No real solution. Juggluco has something similar in 4.8.2, but I don't know if it works because I don't have Android 13. 
So, don't upgrade to Android 13.

**Reply by JulGer** [reply]
Don't really get what you're saying, but thanks. I have deactivated automatic updates anyways. 😉 It all comes down to, (for me, that is) if the developers of MIUI 14 (having a redmi note 10 pro myself) used the updated Android source or an older one. Afaik did Google patch the problem already some time ago.

**Comment by j-kaltes** [comment]
It is no real solution. It sets a timer and if after a certain period of time there is no glucose value it reconnects to the sensor in the hope that it will work this time. Normally this is not needed, somewhere something has gone wrong. Maybe because Android 13 has some battery optimization that disables Bluetooth.
`public static final long BLE_CONNECT_RETRY_FREQUENCY = 58000;`
probably means that it is programmed to go off after 58 seconds. But because of other battery optimizations, in reality will go off after 7 minutes.  So you get again a value after 8 minutes.
